src/storage/remote/read.c

```diff
@@ -43,6 +44,54 @@ Macros for function logging
 #define FUNCTION_LOG_STORAGE_READ_REMOTE_FORMAT(value, buffer, bufferSize)                                                         \
     objToLog(value, "StorageReadRemote", buffer, bufferSize)
 
+/***********************************************************************************************************************************
+Clear protocol if the entire file is not read or an error occurs before the read is complete. This is required to clear the
+protocol state so a subsequent command can succeed.
+***********************************************************************************************************************************/
+static void
+storageReadRemoteFreeResource(THIS_VOID)
+{
+    THIS(StorageReadRemote);
+
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(STORAGE_READ_REMOTE, this);
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+
+    // Read if eof has not been reached
+    if (!this->eof)
+    {
+        do
+        {
+            MEM_CONTEXT_TEMP_BEGIN()
+            {
+                PackRead *const read = protocolClientDataGet(this->client);
+                pckReadNext(read);
+
+                // If binary then discard
+                if (pckReadType(read) == pckTypeBin)
+                {
+                    // Free the buffer immediately on read in case there are many buffers to read
+                    bufFree(pckReadBinP(read));
```

Comment:
The very next thing after this is literally MEM_CONTEXT_TEMP_END when going through this code path, isn't it?  Feel like either we don't need to be creating/destroying a temp mem context on every iteration of the loop as we seem to be doing here and instead move the temp mem context outside the loop and keep the bufFree here, or we don't really need the bufFree here because the temp context destroy will handle it.

Reply:
Oops. I copy-pasted this from `storageReadRemote()` and then did not think hard enough about the contexts.

I think it makes sense to keep the temp mem context since PackRead is also holding a copy of the buffer. Performance is not the main goal here but minimizing memory usage seems like a good idea. So, done that way at 7a233a8d0d2.